logger/drain/simple/drain.go

```diff
@@ -0,0 +1,117 @@
+package simple
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"net/url"
+	"sync"
+	"time"
+)
+
+// This determines how often we'll replace an existing connection with a new one.  This ensures we
+// redial, which in turn will ensure any DNS changes affecting where logs are drained to don't
+// go unnoticed for too long.
+const connRefreshInterval = 5 * time.Minute
```

Comment:
How feasible would it be to make the 4 settings starting at this one above user-configurable?